tests/embedding/test_app.py

```diff
@@ -52,3 +54,67 @@ def test_app_version(embedded_app):
     version = embedded_app.version
     assert type(version) is int
     assert version >= 231
+
+
+@pytest.mark.embedding
+def test_warning_message():
+    """Test pythonnet warning of the embedded instance."""
+    venv_name = "pythonnetvenv"
+    base = os.getcwd()
+
+    if "win" in sys.platform:
+        exe_dir = "Scripts"
+    else:
+        exe_dir = "bin"
+
+    venv_bin = os.path.join(base, "." + venv_name, exe_dir)
+
+    # Set up path to use the virtual environment
+    original_path = os.environ["PATH"]
+    os.environ["PATH"] = venv_bin + os.pathsep + os.environ.get("PATH", "")
+
+    # Create virtual environment
+    subprocess.run([sys.executable, "-m", "venv", "." + venv_name])
+
+    # Upgrade pip
+    print("Upgrading pip")
+    upgrade_pip = subprocess.Popen(
+        [os.path.join(venv_bin, "python"), "-m", "pip", "install", "-U", "pip"]
+    )
+    upgrade_pip.wait()
+
+    # Install tests
+    print("Installing tests")
+    install_tests = subprocess.Popen([os.path.join(venv_bin, "pip"), "install", "-e", ".[tests]"])
+    install_tests.wait()
+
+    # Install pythonnet
+    print("Installing pythonnet")
+    install_pythonnet = subprocess.Popen([os.path.join(venv_bin, "pip"), "install", "pythonnet"])
+    install_pythonnet.wait()
+
+    # Run embedded instance in virtual env with pythonnet installed
+    print("Running the embedded instance in a virtual environment")
+    embedded_py = os.path.join(base, "tests", "scripts", "run_embedded_app.py")
+    check_warning = subprocess.Popen(
+        [os.path.join(venv_bin, "python"), embedded_py], stderr=subprocess.PIPE
+    )
+    check_warning.wait()
+    stderr_output = check_warning.stderr.read().decode()
+
+    """If UserWarning & pythonnet are in the stderr output,
+    set warning to 1. Otherwise, set warning to 0"""
+    if "UserWarning" and "pythonnet" in stderr_output:
+        warning = 1
+    else:
+        warning = 0
```

Comment:
You could also do this:
```suggestion
    warning = True if "UserWarning" and "pythonnet" in stderr_output else False
```